### DISCUSSION TITLE
Solving problems with custom Quarkus extensions

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#35043] [Quarkus Blog/Website]
# Solving problems with custom Quarkus extensions

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/solving-problems-with-custom-extensions/

**Comment by Teja186** [comment]
Hi, 
I am trying to access hapi fhir resources using these depedencies:
 <dependency>
      <groupId>ca.uhn.hapi.fhir</groupId>
      <artifactId>hapi-fhir-client</artifactId>
      <version>${hapi.fhir.version}</version>
    </dependency>
 <dependency>
      <groupId>ca.uhn.hapi.fhir</groupId>
      <artifactId>hapi-fhir-base</artifactId>
      <version>${hapi.fhir.version}</version>
    </dependency>
 <dependency>
      <groupId>org.fresenius</groupId>
      <artifactId>hapi-fhir-dstu3-extension</artifactId>
      <version>1.0.0-SNAPSHOT</version>
    </dependency>

I am able to run the app locally but when I deploy this on openshift serverless, knative, I am getting this error:
exception: java.lang.IllegalStateException: Could not find the HAPI-FHIR structure JAR on the classpath for version DSTU3. Note that as of HAPI-FHIR v0.8, a separate FHIR strcture JAR must be added to your classpath (or project pom.xml if you are using Maven)
	at ca.uhn.fhir.context.FhirContext.<init>(FhirContext.java:150)
	at ca.uhn.fhir.context.FhirContext.<init>(FhirContext.java:142)
	at ca.uhn.fhir.context.FhirContext.forDstu3(FhirContext.java:914)
	at me.daniel.service.HTService.processHomeTreatment(HTService.java:60)
	at me.daniel.service.HTService.consume(HTService.java:39)
	at me.daniel.service.HTService_ClientProxy.consume(HTService_ClientProxy.zig:187)
	at 
Do we have to declare this as an extension by writing custom code similar to the one above you showed?
can we have all the hhir packages inlcuded in one extension?
can you pease suggest. My Architecture team is trying to implement serverless and I am working on a POC. if I can hook up fhir dependencies successfully we can go with qurakus, knative, pretty much for everything we need fhir.

Thanks,
Teja

**Reply by Teja186** [reply]
I mentioned wrong set of dependencies in my question. These are the dependencies we are using in our app:

hapi.fhir.version =5.0.0

 <dependency>
      <groupId>ca.uhn.hapi.fhir</groupId>
      <artifactId>hapi-fhir-base</artifactId>
      <version>${hapi.fhir.version}</version>
    </dependency>

 <dependency>
      <groupId>ca.uhn.hapi.fhir</groupId>
      <artifactId>hapi-fhir-structures-dstu3</artifactId>
      <version>${hapi.fhir.version}</version>
    </dependency>

 <dependency>
      <groupId>ca.uhn.hapi.fhir</groupId>
      <artifactId>hapi-fhir-client</artifactId>
      <version>${hapi.fhir.version}</version>
</dependecy>
